keyservice/internal/server/server.go

```diff
@@ -82,3 +85,27 @@ func (s *Server) GetDataKey(ctx context.Context, in *keyserviceproto.GetDataKeyR
 	}
 	return &keyserviceproto.GetDataKeyResponse{DataKey: key}, nil
 }
+
+// GetCAKey returns the CA key.
+func (s *Server) GetCAKey(ctx context.Context, in *keyserviceproto.GetCAKeyRequest) (*keyserviceproto.GetCAKeyResponse, error) {
+	log := s.log.With("peerAddress", grpclog.PeerAddrFromContext(ctx))
+
+	key, err := s.conKMS.GetDEK(ctx, crypto.DEKPrefix, 256)
+	if err != nil {
+		log.With(slog.Any("error", err)).Error("Failed to get key for derivation")
+		return nil, status.Errorf(codes.Internal, "%v", err)
+	}
+
+	_, priv, err := ed25519.GenerateKey(bytes.NewReader(key))
```

Comment:
Does the keyservice really need to know about SSH? We could keep most interfaces as they are if we just used `GetDataKey` and derived the SSH public key in the joinclient.